### ISSUE TITLE
Custom shortcuts cause crash dump

### ISSUE THREAD

**Issue by jauntyd** [quodlibet/quodlibet#3409]
Hello:
I enjoy using the program, however, when I set custom keyboard shortcuts in XFCE4 on FreeBSD 12.1-STABLE I get the follow crash dump:

=== SYSTEM INFORMATION:
Quod Libet 3.6.2
Mutagen 1.42.0
Python 2.7.18 (default, Jul  7 2020, 01:20:02) 
[GCC 4.2.1 Compatible FreeBSD Clang 8.0.1 (tags/RELEASE_801/final 366581)] freebsd12
Platform FreeBSD-12.1-STABLE-amd64-64bit-ELF

=== STACK TRACE
Traceback (most recent call last):

  File "/usr/local/lib/python2.7/site-packages/gi/overrides/GLib.py", line 668, in <lambda>
    func_fdtransform = lambda _, cond, *data: callback(channel, cond, *data)

  File "/usr/local/lib/python2.7/site-packages/quodlibet/util/fifo.py", line 233, in _process
    self._open(*args)

TypeError: _open() takes at least 2 arguments (1 given)

=== LOG:
D: 898.71: PluginManager.register_handler: Registering handler: 'GStreamerPlayer'
D: 898.71: PluginManager.register_handler: Registering handler: 'SongsMenuPluginHandler'
D: 898.71: PluginManager.register_handler: Registering handler: 'CoverPluginHandler'
D: 898.71: PluginManager.register_handler: Registering handler: 'PlaylistPluginHandler'
D: 898.71: PluginManager.register_handler: Registering handler: 'QueryPluginHandler'
D: 898.81: PluginManager.register_handler: Registering handler: 'PlayOrder'
D: 898.82: GStreamerPlayer.go_to: Going to None
D: 898.82: PlaylistMux.go_to: Told to go to None
D: 898.82: QueueModel.go_to: Told to go to None
D: 898.82: PlaylistModel.go_to: Told to go to None
D: 898.82: AlbumLibrary.__init__: Initializing Album Library to watch 'main'
D: 898.85: CoPool.add: Added copool function <bound method SongFileLibrary.rebuild of <libraries.SongFileLibrary object at 0x8053f8550 (quodlibet+library+libraries+SongFileLibrary at 0x8045a3b60)>> with id 'library'
D: 898.85: qltk.window.__restore_state: Restore state
D: 898.85: qltk.window.__restore_position: Restore position
D: 898.85: qltk.window.__restore_size: Restore size
D: 898.85: PluginManager.register_handler: Registering handler: 'EventPluginHandler'
D: 898.85: mmkeys.find_active_backend: Trying to find a mmkeys backend
D: 898.85: FIFO._open: fifo locking failed: IOError(45, 'Operation not supported')
D: 898.86: qltk.session.init: Connecting with the gnome session manager failed
D: 898.86: CoPool.add: Added copool function <function periodic_config_save at 0x806fbb550> with id <function periodic_config_save at 0x806fbb550>
D: 898.86: CoPool.add: Added copool function <function periodic_library_save at 0x806fbb3d0> with id <function periodic_library_save at 0x806fbb3d0>
D: 898.86: quodlibet.main: Entering quodlibet.main
D: 898.92: PlaylistModel.set: Clearing model.
D: 898.92: PlaylistModel.set: Setting 9 songs.
D: 898.92: PlaylistModel.set: Done filling model.
D: 898.92: GStreamerPlayer.go_to: Going to '/usr/home/john/music/flac/Iskra String Quartet/Iskra/03 Melodia (I).flac'
D: 898.92: PlaylistMux.go_to: Told to go to '/usr/home/john/music/flac/Iskra String Quartet/Iskra/03 Melodia (I).flac'
D: 898.92: QueueModel.go_to: Told to go to None
D: 898.92: PlaylistModel.go_to: Told to go to '/usr/home/john/music/flac/Iskra String Quartet/Iskra/03 Melodia (I).flac'
D: 898.92: GStreamerPlayer._end: End song
D: 898.92: util.cover.manager.failure: Didn't get cover from FilesystemCover: This source is incapable of fetching covers
D: 898.96: util.cover.manager.run: Found local cover from EmbedCover
D: 898.97: GStreamerPlayer._end: Next song
D: 899.08: player.gstbe.player.__message: Recalculate latency
D: 899.09: player.gstbe.player.__message: Recalculate latency
D: 899.17: CoPool.add: Added copool function <bound method AlbumList.__scan_paths of <main.AlbumList object at 0x806cdec30 (quodlibet+browsers+albums+main+AlbumList at 0x806ea6570)>> with id <bound method AlbumList.__scan_paths of <main.AlbumList object at 0x806cdec30 (quodlibet+browsers+albums+main+AlbumList at 0x806ea6570)>>
D: 899.57: quodlibet._override_exceptions: Enabling custom exception handler.
D: 899.61: util.set_process_title: Couldn't find module libc.so.6 (ctypes). Not setting process title.
D: 899.61: SongFileLibrary.rebuild: Rebuilding, force is False.
D: 899.61: SongFileLibrary._changed: Changing 1 items.
D: 899.61: library.libraries.__changed: Updating affected albums for 1 items
D: 899.61: CoPool.remove: Removed copool function id <bound method AlbumList.__scan_paths of <main.AlbumList object at 0x806cdec30 (quodlibet+browsers+albums+main+AlbumList at 0x806ea6570)>>
D: 899.61: CoPool.add: Added copool function <bound method AlbumList.__scan_paths of <main.AlbumList object at 0x806cdec30 (quodlibet+browsers+albums+main+AlbumList at 0x806ea6570)>> with id <bound method AlbumList.__scan_paths of <main.AlbumList object at 0x806cdec30 (quodlibet+browsers+albums+main+AlbumList at 0x806ea6570)>>
D: 899.69: SongFileLibrary.rebuild: Removing 0, changing 0.
D: 899.69: SongFileLibrary.scan: Scanning '/usr/home/john/music/flac'.
D: 899.70: CoPool.remove: Removed copool function id 'library'
D: 899.70: CoPool.remove: Removed copool function id <bound method AlbumList.__scan_paths of <main.AlbumList object at 0x806cdec30 (quodlibet+browsers+albums+main+AlbumList at 0x806ea6570)>>
D: 915.32: CoPool.add: Added copool function <bound method GObjectMeta.__build_model of <class 'quodlibet.qltk.completion.LibraryTagCompletion'>> with id <bound method GObjectMeta.__build_model of <class 'quodlibet.qltk.completion.LibraryTagCompletion'>>
D: 915.42: qltk.completion.__build_model: Updating tag model for whole library
D: 915.44: qltk.completion.__build_model: Done updating tag model for whole library
D: 915.44: CoPool.remove: Removed copool function id <bound method GObjectMeta.__build_model of <class 'quodlibet.qltk.completion.LibraryTagCompletion'>>
D: 926.52: PlaylistModel.set: Clearing model.
D: 926.52: PlaylistModel.set: Setting 9 songs.
D: 926.52: PlaylistModel.set: Done filling model.
D: 935.60: util.collection.__populate_from_file: Playlist 'Mellow' not found, creating new.
D: 942.14: PlaylistModel.set: Clearing model.
D: 942.14: PlaylistModel.set: Setting 9 songs.
D: 942.14: PlaylistModel.set: Done filling model.
D: 942.32: QueueModel.go_to: Told to go to None
D: 942.32: PlaylistModel.go_to: Told to go to None
D: 942.32: GStreamerPlayer._end: End song
D: 942.32: util.cover.manager.failure: Didn't get cover from FilesystemCover: This source is incapable of fetching covers
D: 942.33: util.cover.manager.run: Found local cover from EmbedCover
D: 942.33: GStreamerPlayer._end: Next song
D: 942.41: player.gstbe.player.__message: Recalculate latency
D: 942.41: player.gstbe.player.__message: Recalculate latency
D: 942.41: SongFileLibrary._changed: Changing 2 items.
D: 942.41: qltk.completion.__update_song: Updating tag model for 2 songs
D: 942.41: qltk.completion.__update_song: Done updating tag model for 2 songs
D: 942.41: library.libraries.__changed: Updating affected albums for 2 items
D: 952.18: CommandRegistry.handle_line: command: play-pause(*[])
D: 952.18: CommandRegistry.run: Running <function _play_pause at 0x806facc50> with params () 
D: 952.18: FIFO._open: fifo locking failed: IOError(45, 'Operation not supported')
D: 953.57: CommandRegistry.handle_line: command: play-pause(*[])
D: 953.57: CommandRegistry.run: Running <function _play_pause at 0x806facc50> with params () 

I set f1 to "quodlibet --play-pause"

Thanks,

**Comment by Meriipu** [comment]
3.6.2 is a pretty old version

**Comment by jauntyd** [comment]
I will hunt down source or a newer version. thanks.

**Comment by Meriipu** [comment]
for what it is worth, I have using xbindkeys to do `quodlibet --play-pause` on linux since around the start of 2017 to play/pause. I think the version I used back then was 3.8 or something. 
It is up to 4.3.0 now (+ some commits on top of that) and has worked for me during all this time, so perhaps it was fixed in between 3.6 and 3.8 or 3.9? 

(of course it might also be a setup-specific issue that still happens on a recent version)
